migration_scripts/bulk_load/src/footprint.clj

```diff
@@ -73,43 +73,91 @@
-         ;; which (potentially partitioned) table it belongs to.
-         (map (fn [{:pg_class/keys [relname relkind reltuples]
-                    :keys [tab_size rel_size]}]
-                [(table->entity relname)
-                 (cond
-                   (= relkind "r")
-                   {:self rel_size
-                    :toast (- tab_size rel_size)
-                    :tuples reltuples}
-                   (s/ends-with? relname "_pkey")
-                   {:pkey rel_size}
-                   :else
-                   {:idx rel_size})]))
-
-         ;; Group by the table name, and merge entries together, so that for
-         ;; each table we get a single stats map.
+            (partitioned :transactions)))
+
+        per-table-size-stats
+        (->> ["SELECT
+                   relname,
+                   relkind,
+                   reltuples,
+                   pg_table_size(CAST(relname AS VARCHAR)) tab_size,
+                   pg_relation_size(CAST(relname AS VARCHAR)) rel_size
+               FROM
+                   pg_class
+               WHERE
+                   relkind IN ('i', 'r')
+               AND relname NOT LIKE 'pg_%'
+               AND relname NOT LIKE 'sql_%'
+               AND relname NOT LIKE '__diesel_%'"]
+             (jdbc/execute! db)
+
+             ;; Each row represents a table or an index. Figure out
+             ;; which (potentially partitioned) table it belongs to.
+             (map (fn [{:pg_class/keys [relname relkind reltuples]
+                        :keys [tab_size rel_size]}]
+                    [(table->entity relname)
+                     (cond
+                       (= relkind "r")
+                       {:self rel_size
+                        :toast (- tab_size rel_size)
+                        :tuples reltuples}
+                       (s/ends-with? relname "_pkey")
+                       {:pkey rel_size}
+                       :else
+                       {:idx rel_size})]))
+
+             ;; Group by the table name, and merge entries together, so that for
+             ;; each table we get a single stats map.
+             (group-by first)
+             (map (fn [[key stats]]
+                    [key (apply merge-with +
+                                (map second stats))])))
+
+        ;; Note that column stats may be missing if the relevant
+        ;; tables haven't been analysed.
+        per-table-column-stats
+        (->> ["SELECT
+                   tablename,
+                   attname,
+                   avg_width
+               FROM
+                   pg_stats
+               WHERE
+                   tablename NOT LIKE 'pg_%'
+               AND tablename NOT LIKE 'sql_%'
+               AND tablename NOT LIKE '__diesel_%'
+               AND tablename NOT IN (
+                   'events',
+                   'transactions',
+                   'objects_history'
+               )"]
```

Comment:
It's actually the other way around -- we generate column stats for every partition of a partitioned table individually (and it's useful to do that because then we get an average per partition rather than as a whole), and we're discarding stats about the overall partitioned table, because they would be at best a summary of the information we're already gathering.